packages/react/src/components/FileUploader/FileUploaderButton.tsx

```diff
@@ -13,9 +13,98 @@ import { ButtonKinds } from '../../prop-types/types';
 import uid from '../../tools/uniqueId';
 import { usePrefix } from '../../internal/usePrefix';
 import deprecate from '../../prop-types/deprecate';
+import { ReactAttr } from '../../types/common';
 
 function noop() {}
 
+export interface FileUploaderButtonProps
+  extends Omit<ReactAttr<HTMLButtonElement>, 'onChange' | 'tabIndex'> {
+  /**
+   * Specify the types of files that this input should be able to receive
+   */
+  accept?: string[];
+
+  /**
+   * Specify the type of underlying button
+   */
+  buttonKind?:
+    | 'primary'
+    | 'secondary'
+    | 'danger'
+    | 'ghost'
+    | 'danger--primary'
+    | 'danger--ghost'
+    | 'danger--tertiary'
+    | 'tertiary';
+
+  /**
+   * Provide a custom className to be applied to the container node
+   */
+  className?: string;
+
+  /**
+   * Specify whether you want to disable any updates to the FileUploaderButton
+   * label
+   */
+  disableLabelChanges?: boolean;
+
+  /**
+   * Specify whether file input is disabled
+   */
+  disabled?: boolean;
+
+  /**
+   * Provide a unique id for the underlying `<input>` node
+   */
+  id?: string;
+
+  /**
+   * Provide the label text to be read by screen readers when interacting with
+   * this control
+   */
+  labelText?: React.ReactNode;
+
+  /**
+   * Specify if the component should accept multiple files to upload
+   */
+  multiple?: boolean;
+
+  /**
+   * Provide a name for the underlying `<input>` node
+   */
+  name?: string;
+
+  /**
+   * Provide an optional `onChange` hook that is called each time the `<input>`
+   * value changes
+   */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+
+  /**
+   * Provide an optional `onClick` hook that is called each time the button is
+   * clicked
+   */
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+
+  /**
+   * Provide an accessibility role for the `<FileUploaderButton>`
+   */
+  role?: string;
+
+  /**
+   * Specify the size of the FileUploaderButton, from a list of available
+   * sizes.
+   */
+  size?: 'sm' | 'small' | 'field' | 'md' | 'lg';
+
+  /**
+   * @deprecated The `tabIndex` prop for `FileUploaderButton` has been deprecated since it now renders a button element by default.
+   */
+  tabIndex?: number | string;
```

Comment:
Could you share more of why this was updated to accept a string? It should only be a number IIRC

Reply:
You can read about that here: https://github.com/carbon-design-system/carbon/pull/13992#issuecomment-1589923529 - but essentially it used to be a string in the proptypes (you can see that in the snapshot diff). I guess it has worked previously just because of how HTML stringifies certain props if given as different types.  

I agree that it should really only be a number. Typescript should yell at them a few places if they've used strings previously, and then it should be a pretty easy refactor to fix in that case. 

But that's not a decicion for me to make, so I guess it's up to you guys to decide how you want to handle this change @tay1orjones & @tw15egan. Let me know where you land and I'll update the PR to match it! 🚀

Reply:
Ah sorry I missed that, GitHub auto-collapsed that part of the conversation and I didn't look into it. I think it's fine as is for now